modules/javafx.graphics/src/main/native-glass/gtk/glass_general.cpp

```diff
@@ -584,17 +584,23 @@ glass_gdk_mouse_devices_ungrab() {
 }
 
 void
-glass_gdk_master_pointer_grab(GdkWindow *window, GdkCursor *cursor) {
+glass_gdk_master_pointer_grab(GdkEvent *event, GdkWindow *window, GdkCursor *cursor) {
     if (disableGrab) {
         gdk_window_set_cursor(window, cursor);
         return;
     }
 #ifdef GLASS_GTK3
-        gdk_device_grab(gdk_device_manager_get_client_pointer(
-                    gdk_display_get_device_manager(
-                        gdk_display_get_default())),
-                    window, GDK_OWNERSHIP_NONE, FALSE, GDK_ALL_EVENTS_MASK,
-                    cursor, GDK_CURRENT_TIME);
+        GdkDevice *device = gdk_event_get_device(event);
+
+        gdk_device_grab(device, window, GDK_OWNERSHIP_NONE, FALSE,
+                        (GdkEventMask)
+                             (GDK_POINTER_MOTION_MASK
+                                 | GDK_BUTTON_MOTION_MASK
+                                 | GDK_BUTTON1_MOTION_MASK
+                                 | GDK_BUTTON2_MOTION_MASK
+                                 | GDK_BUTTON3_MOTION_MASK
+                                 | GDK_BUTTON_RELEASE_MASK),
+                           cursor, GDK_CURRENT_TIME);
 #else
```

Comment:
Is there any specific reason to change GDK_ALL_EVENTS_MASK to these specific events?

Reply:
Yes. Those are the same events used on gtk2, so I thought only to grab the events needed. Since gtk2 DND is well tested, I thought it's the best choice of events.